src/postgres-util/src/lib.rs

```diff
@@ -111,3 +112,41 @@ pub async fn table_info(conn: &str, ast_table: &String) -> Result<TableInfo, any
 
     Ok(TableInfo { rel_id, schema })
 }
+
+pub async fn drop_replication_slots(conn: &str, slots: &[String]) -> Result<(), anyhow::Error> {
+    let (client, connection) = tokio_postgres::connect(&conn, NoTls).await?;
+    tokio::spawn(connection);
+
+    for slot in slots {
+        let (active, active_pid) = query_pg_replication_slots(&client, slot).await?;
+        if active {
+            client
+                .query(
+                    "SELECT pg_terminate_backend($1)",
+                    &[&active_pid.expect("active backend task without pid")],
```

Comment:
Since we never expect `active` to be `true` and `active_pid` to be `None` I'm thinking if we could just return the `Option<i32>` from `query_pg_replication_slots` and terminate the backend by matching on that